crates/modelardb_compression/src/types.rs

```diff
@@ -555,58 +557,263 @@ mod tests {
 
     // Tests for CompressedSegmentBuilder.
     #[test]
-    fn test_model_attributes_for_pmc_mean() {
-        let uncompressed_timestamps = TimestampArray::from(UNCOMPRESSED_TIMESTAMPS.to_vec());
+    fn test_encoding_decoding_for_pmc_mean_without_residuals() {
         let uncompressed_values = ValueArray::from(vec![10.0, 10.0, 10.0, 10.0, 10.0]);
+        test_encoding_decoding_for_pmc_mean(uncompressed_values, 4, 10.0, 0, 10.0, 10.0, 0);
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_pmc_mean_with_residuals_with_new_mininum() {
+        let uncompressed_values = ValueArray::from(vec![10.0, 10.0, 10.0, 10.0, Value::MIN]);
+        test_encoding_decoding_for_pmc_mean(uncompressed_values, 3, 10.0, 0, Value::MIN, 10.0, 1);
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_pmc_mean_with_residuals_with_new_maximum() {
+        let uncompressed_values = ValueArray::from(vec![10.0, 10.0, 10.0, 10.0, Value::MAX]);
+        test_encoding_decoding_for_pmc_mean(uncompressed_values, 3, 10.0, 0, 10.0, Value::MAX, 0);
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_pmc_mean_with_residuals_with_new_minimum_and_maximum() {
+        let uncompressed_values = ValueArray::from(vec![10.0, 10.0, 10.0, Value::MIN, Value::MAX]);
+        test_encoding_decoding_for_pmc_mean(
+            uncompressed_values,
+            2,
+            10.0,
+            0,
+            Value::MIN,
+            Value::MAX,
+            4,
+        );
+    }
+
+    fn test_encoding_decoding_for_pmc_mean(
```

Comment:
I am not sure this is a convention we currently use consistently but maybe we should attempt to avoid naming test utility functions such as this with the "test_" prefix and only use this prefix for actual tests. Utility functions that use `assert!` could maybe be named `assert_` instead.

Reply:
Fixed.